### ISSUE TITLE
ofNodeExample shader error on OSX

### ISSUE THREAD

**Issue by edap** [openframeworks/openFrameworks#5383]
Hi,
When running the ofNodeExample from the last nigthly release (2/12/2016) I've the following error:

```bash
[ error ] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader failed to compile
[ error ] ofShader: GL_VERTEX_SHADER shader reports:
ERROR: 0:14: '' : syntax error: incorrect preprocessor directive
ERROR: 0:14: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
ERROR: 0:40: '' : syntax error: incorrect preprocessor directive
ERROR: 0:40: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline

[ error ] ofShader: GL_VERTEX_SHADER, offending line 14 :
	   12	OUT vec3 v_eyePosition;
	   13	OUT vec3 v_worldPosition;
	   14	#if HAS_COLOR
	   15	OUT vec4 v_color;
	   16	#endif

[ error ] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader failed to compile
[ error ] ofShader: GL_FRAGMENT_SHADER shader reports:
ERROR: 0:17: '' : syntax error: incorrect preprocessor directive
ERROR: 0:17: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
ERROR: 0:246: '' : syntax error: incorrect preprocessor directive
ERROR: 0:246: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline

[ error ] ofShader: GL_FRAGMENT_SHADER, offending line 17 :
	   15	    IN vec3 v_eyePosition;
	   16	    IN vec3 v_worldPosition;
	   17	#if HAS_COLOR
	   18	    IN vec4 v_color;
	   19	#endif

[ error ] ofShader: checkProgramLinkStatus(): program failed to link
[ error ] ofShader: ofShader: program reports:
ERROR: One or more attached shaders not successfully compiled

[ error ] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader failed to compile
[ error ] ofShader: GL_VERTEX_SHADER shader reports:
ERROR: 0:15: '' : syntax error: incorrect preprocessor directive
ERROR: 0:15: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
ERROR: 0:41: '' : syntax error: incorrect preprocessor directive
ERROR: 0:41: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline

[ error ] ofShader: GL_VERTEX_SHADER, offending line 15 :
	   13	OUT vec3 v_eyePosition;
	   14	OUT vec3 v_worldPosition;
	   15	#if HAS_COLOR
	   16	OUT vec4 v_color;
	   17	#endif

[ error ] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader failed to compile
[ error ] ofShader: GL_FRAGMENT_SHADER shader reports:
ERROR: 0:18: '' : syntax error: incorrect preprocessor directive
ERROR: 0:18: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
ERROR: 0:247: '' : syntax error: incorrect preprocessor directive
ERROR: 0:247: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline

[ error ] ofShader: GL_FRAGMENT_SHADER, offending line 18 :
	   16	    IN vec3 v_eyePosition;
	   17	    IN vec3 v_worldPosition;
	   18	#if HAS_COLOR
	   19	    IN vec4 v_color;
	   20	#endif

[ error ] ofShader: checkProgramLinkStatus(): program failed to link
[ error ] ofShader: ofShader: program reports:
ERROR: One or more attached shaders not successfully compiled

[ error ] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader failed to compile
[ error ] ofShader: GL_VERTEX_SHADER shader reports:
ERROR: 0:15: '' : syntax error: incorrect preprocessor directive
ERROR: 0:15: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
ERROR: 0:41: '' : syntax error: incorrect preprocessor directive
ERROR: 0:41: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline

[ error ] ofShader: GL_VERTEX_SHADER, offending line 15 :
	   13	OUT vec3 v_eyePosition;
	   14	OUT vec3 v_worldPosition;
	   15	#if HAS_COLOR
	   16	OUT vec4 v_color;
	   17	#endif

[ error ] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader failed to compile
[ error ] ofShader: GL_FRAGMENT_SHADER shader reports:
ERROR: 0:18: '' : syntax error: incorrect preprocessor directive
ERROR: 0:18: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
ERROR: 0:247: '' : syntax error: incorrect preprocessor directive
ERROR: 0:247: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline

[ error ] ofShader: GL_FRAGMENT_SHADER, offending line 18 :
	   16	    IN vec3 v_eyePosition;
	   17	    IN vec3 v_worldPosition;
	   18	#if HAS_COLOR
	   19	    IN vec4 v_color;
	   20	#endif

[ error ] ofShader: checkProgramLinkStatus(): program failed to link
[ error ] ofShader: ofShader: program reports:
ERROR: One or more attached shaders not successfully compiled

[ error ] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader failed to compile
[ error ] ofShader: GL_FRAGMENT_SHADER shader reports:
ERROR: 0:247: '' : syntax error: incorrect preprocessor directive
ERROR: 0:247: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline

[ error ] ofShader: GL_FRAGMENT_SHADER, offending line 247 :
	  245	        ////////////////////////////////////////////////////////////
	  246	        // now add the material info
	  247	        #if HAS_TEXTURE && !HAS_COLOR
	  248	            vec4 tex = TEXTURE(tex0, v_texcoord);
	  249	            vec4 localColor = vec4(ambient,1.0) * tex + vec4(diffuse,1.0) * tex + vec4(specular,1.0) * mat_specular + mat_emissive;

[ error ] ofShader: checkProgramLinkStatus(): program failed to link
[ error ] ofShader: ofShader: program reports:
ERROR: One or more attached shaders not successfully compiled


```

I'm on OSX el Capitan, xcode 7.2

**Comment by edap** [comment]
This example is working in OF 0.9.8

**Comment by bakercp** [comment]
Closing as it is fixed in 0.9.8.
